connection.js

```diff
@@ -459,6 +459,65 @@ class Connection {
+            }
+            // then, after eliminating > 9 rank difference if ranked, we consider value of min-max handicap if set
+            // we eliminate all unwanted values, everything not forbidden is allowed
+            if (config.minhandicap && (rankDifference < config.minhandicap)) {
+                conn_log("Automatic handicap " + rankDifference + " stones is too low, minimum handicap stone number is " + config.minhandicap);
+                return {reject: true, msg: "Automatic handicap " + rankDifference + " stones is too low, minimum handicap stone number is " + config.minhandicap + ", please increase the number of handicap stones in -custom handicap-"};
+            }
+            if (config.minhandicapranked && notification.ranked && (rankDifference < config.minhandicapranked)) {
+                conn_log("Automatic handicap " + rankDifference + " stones is too low, minimum handicap stone number for ranked games is " + config.minhandicapranked);
+                return {reject: true, msg: "Automatic handicap " + rankDifference + " stones is too low, minimum handicap stone number for ranked games is " + config.minhandicapranked + ", please increase the number of handicap stones in -custom handicap-"};
+            }
+            if (config.minhandicapunranked && !notification.ranked && (rankDifference < config.minhandicapunranked)) {
+                conn_log("Automatic handicap " + rankDifference + " stones is too low, minimum handicap stone number for unranked games is " + config.minhandicapunranked);
+                return {reject: true, msg: "Automatic handicap " + rankDifference + " stones is too low, minimum handicap stone number for unranked games is " + config.minhandicapunranked + ", please increase the number of handicap stones in -custom handicap-"};
+            }
+            if (config.maxhandicap && (rankDifference > config.maxhandicap)) {
+                conn_log("Automatic handicap " + rankDifference + " stones is too high, maximum handicap stone number is " + config.maxhandicap);
+                return {reject: true, msg: "Automatic handicap " + rankDifference + " stones is too high, maximum handicap stone number is " + config.maxhandicap + ", please reduce the number of handicap stones in -custom handicap-"};
+            }
+            if (config.maxhandicapranked && notification.ranked && (rankDifference > config.maxhandicapranked)) {
+                conn_log("Automatic handicap " + rankDifference + " stones is too high, maximum handicap stone number for ranked games is " + config.maxhandicapranked);
+                return {reject: true, msg: "Automatic handicap " + rankDifference + " stones is too high, maximum handicap stone number for ranked games is " + config.maxhandicapranked + ", please reduce the number of handicap stones in -custom handicap-"};
+            }
+            if (config.maxhandicapunranked && !notification.ranked && (rankDifference > config.maxhandicapunranked)) {
```

Comment:
I think you should figure out some way of reducing the repetition here.

Reply:
maybe we can start with this : https://github.com/online-go/gtp2ogs/pull/165#discussion_r263528457